superset/assets/javascripts/explore/components/controls/AnnotationLayer.jsx

```diff
@@ -0,0 +1,558 @@
+          this.setState({
+            isLoadingOptions: false,
+            valueOptions: layers,
+          });
+        });
+      } else if (isQueryAnnotationType(annotationType)) {
+        $.ajax({
+          type: 'GET',
+          url: '/superset/user_slices',
+        }).then(data =>
+          this.setState({
+            isLoadingOptions: false,
+            valueOptions: data.filter(x => supportedSliceTypes(annotationType)
+              .find(v => v === x.viz_type))
+              .map(x => ({ value: x.id, label: x.title, slice: x })),
+          }),
+        );
+      }
+    }
+  }
+
+  deleteAnnotation() {
+    this.props.close();
+    if (!this.state.isNew) {
+      this.props.removeAnnotationLayer(this.state);
+    }
+  }
+
+  applyAnnotation() {
+    if (this.state.name.length) {
+      const annotation = { ...this.state };
+      annotation.color = annotation.color === AUTOMATIC_COLOR ? null : annotation.color;
+      delete annotation.isNew;
+      delete annotation.valueOptions;
+      delete annotation.isLoadingOptions;
+      this.props.addAnnotationLayer(annotation);
+      this.setState({ isNew: false, oldName: this.state.name });
+    }
+  }
+
+  submitAnnotation() {
+    this.applyAnnotation();
+    this.props.close();
+  }
+
+  renderValueConfiguration() {
+    const { annotationType, value, valueOptions, isLoadingOptions } = this.state;
+    let label = '';
+    let description = '';
+    if (annotationType === AnnotationTypes.NATIVE) {
+      label = 'Annotation Layer';
+      description = 'Select the Annotation Layer you would like to use.';
+    } else if (isQueryAnnotationType(annotationType)) {
+      label = 'Slice';
+      description = 'Use a pre defined Superset Slice as a source for annotations and overlays. ' +
+        'your Slice must be one of these visualization types: ' +
+        `[${supportedSliceTypes(annotationType).map(x => vizTypes[x].label).join(', ')}]`;
+    } else if (annotationType === AnnotationTypes.FORMULA) {
+      label = 'Formula';
+      description = 'Expects a formula with depending time parameter `x`' +
+        ' in milliseconds since epoch. mathjs is used to evaluate the formulas.';
+    }
+    if (isQueryAnnotationType(annotationType) || annotationType === AnnotationTypes.NATIVE) {
+      return (
+        <SelectControl
+          name="annotation-layer-value"
+          showHeader
+          hovered
+          description={description}
+          label={label}
+          placeholder=""
+          options={valueOptions}
+          isLoading={isLoadingOptions}
+          value={value}
+          onChange={this.handleValue}
+          validationErrors={!value ? ['Mandatory'] : []}
+        />
+      );
+    } if (annotationType === AnnotationTypes.FORMULA) {
+      return (
+        <TextControl
+          name="annotation-layer-value"
+          hovered
+          showHeader
+          description={description}
+          label={label}
+          placeholder=""
```

Comment:
would it be useful to give a simple example here? like `2*x + 5`?

Reply:
that makes sense re the default, I think an example would still be useful though so users understand the expected format.